src/rendering/SceneManager.cc

```diff
@@ -848,6 +849,98 @@ rendering::VisualPtr SceneManager::CreateActor(Entity _id,
   return actorVisual;
 }
 
+/////////////////////////////////////////////////
+rendering::VisualPtr SceneManager::CreateLightVisual(Entity _id,
+    const sdf::Light &_light, Entity _parentId)
+{
+  if (!this->dataPtr->scene)
+    return rendering::VisualPtr();
+
+  if (this->dataPtr->visuals.find(_id) != this->dataPtr->visuals.end())
+  {
+    ignerr << "Entity with Id: [" << _id << "] already exists in the scene"
+           << std::endl;
+    return rendering::VisualPtr();
+  }
+
+  std::string name = _light.Name().empty() ? std::to_string(_id) :
+      _light.Name();
+
+  rendering::VisualPtr visualParent;
+  rendering::LightPtr lightParent;
+  if (_parentId != this->dataPtr->worldId)
+  {
+    auto it = this->dataPtr->visuals.find(_parentId);
+    if (it != this->dataPtr->visuals.end())
```

Comment:
Shouldn't a light visual always be attached to a light? In which situation would a light visual be attached to another visual? Looking at the code, it looks like `_parentId` will either be the ID of a light or a world.

Reply:
When there is an object with a light inside (like the point light in the sphere of the `light.sdf` example) we need to use this logic, otherwise the light visual is not visualized. We need to add the light visual child to the upper visual, in this case the sphere visual.

Reply:
I started looking at the idea of creating LightVisuals always as a child of light. I think this may work: #608